encrypted-media/scripts/clearkey-update-non-ascii-input.js

```diff
@@ -0,0 +1,44 @@
+// This test is only applicable to clearkey
+function runTest(config, qualifier) {
+    var testname = testnamePrefix(qualifier, config.keysystem) + ' test handling of non-ASCII responses for update()';
+
+    var configuration = getSimpleConfigurationForContent(config.content);
+
+    if (config.initDataType) {
+        configuration.initDataTypes = [config.initDataType];
+    }
+
+    async_test(function (test) {
+        var initDataType;
+        var initData;
+        var mediaKeySession;
+
+        // Clear Key JSON Web Key needs to be ASCII encoded
+        function processMessage(event) {
+                // |jwkSet| contains a  non-ASCII character \uDC00.
+                var jwkSet = '{"keys":[{'
+                    +     '"kty":"oct",'
+                    +     '"k":"MDEyMzQ1Njc4OTAxMjM0NQ",'
+                    +     '"kid":"MDEyMzQ1Njc4O\uDC00TAxMjM0NQ"'
+                    + '}]}';
+
+
+                event.target.update(stringToUint8Array(jwkSet)).then(function () {
+                    forceTestFailureFromPromise(test, 'Error: update() should fail because the processed message has non-ASCII character.');
+                }, function (error) {
```

Comment:
Since here we are still using async_test this should be wrapped with step_func. We may try to convert this one to promise_test too (probably with promise_rejects). In that case forceTestFailureFromPromise won't be necessary.